docs/shape-providers.md

```diff
@@ -202,6 +202,33 @@ abstract record Node(int label, int left, int right) : BinTree
 If left unset, the name of a derived type defaults to its type name (i.e. `nameof(TDerived)`) and the tag corresponds to the attribute declaration order.
 It should be noted that mono reflection does not preserve attribute declaration ordering, so it is recommended that applications targeting mono should either use the source generator or explicitly set the tags for all model types.
 
+For the case of unregistered derived types, PolyType applies a "nearest known ancestor" resolution algorithm. Given the type hierarchy
+
+```C#
+[DerivedTypeShape(typeof(Horse))]
+[DerivedTypeShape(typeof(Cow))]
+class Animal;
+class Horse;
+class Cow;
```

Comment:
Are you missing `: Animal` base types here?

Reply:
Thanks, fixed in https://github.com/eiriktsarpalis/PolyType/pull/124